src/components/close-button/CloseButton.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import Button, { ButtonType } from '../button';
+import IconClose from '../../icons/general/IconClose';
+import { bdlGray62 } from '../../styles/variables';
+
+import './CloseButton.scss';
+
+export interface CloseButtonProps {
+    /** Custom class for the close button */
+    className?: string;
+    /** onClick handler for the close button */
+    onClick?: Function;
+}
+
+const CloseButton = ({ className, onClick }: CloseButtonProps) => {
```

Comment:
Another Typescript pattern that I really like is to use `React.FC` and provide props via a generic:
`const CloseButton: React.FC<CloseButtonProps> = happiness`
I don't know if any official decisions have been made, actually, about how to type React components. Might be a good topic for a very mini RFC.

Speaking of RFC, we may able to replace default exports with named exports in the context of Typescript sometime soon: https://cloud.box.com/s/50b9hr6z5t7213cn79052nfqxn5vhzun . However, that RFC is still unresolved afaik, so default for now; but I thought you'd be interested :)